workspaces/copilot/plugins/copilot-backend/src/client/GithubClient.ts

```diff
@@ -64,17 +67,84 @@ export class GithubClient implements GithubApi {
   private async getOctokit(
     type: 'enterprise' | 'organization',
   ): Promise<Octokit> {
-    const credentials = await this.getCredentials();
-    const headers = credentials[type]?.headers || {};
-
-    return new Octokit({
-      baseUrl: this.copilotConfig.apiBaseUrl,
-      auth: headers.Authorization?.replace('Bearer ', '') || '',
-      headers: {
-        Accept: 'application/vnd.github+json',
-        'X-GitHub-Api-Version': '2022-11-28',
-      },
-    });
+    try {
+      const credentials = await this.getCredentials();
+      const authStrategy = credentials[type];
+
+      if (!authStrategy) {
+        throw new Error(`No credentials configured for ${type}`);
+      }
+
+      const octokitConfig: any = {
+        baseUrl: this.copilotConfig.apiBaseUrl,
+        headers: {
+          Accept: 'application/vnd.github+json',
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+      };
+
+      // If it's a string token, use it directly
+      // If it's an auth strategy config (GitHub App), pass authStrategy and auth
+      if (typeof authStrategy === 'string') {
+        this.logger?.debug(
+          `[GithubClient] Using token authentication for ${type}`,
+        );
+        octokitConfig.auth = authStrategy;
+      } else {
+        this.logger?.debug(
+          `[GithubClient] Using GitHub App authentication for ${type}`,
+          {
+            appId: authStrategy.appId,
+          },
+        );
+
+        // For GitHub Apps, we need to get the installation ID first
+        // Create a temporary app-authenticated Octokit to get the installation ID
+        const appOctokit = new Octokit({
+          authStrategy: createAppAuth,
+          auth: {
+            appId: authStrategy.appId,
+            privateKey: authStrategy.privateKey,
+          },
+        });
+
+        // Get installation ID for the organization
+        const orgName =
+          type === 'organization'
+            ? this.copilotConfig.organization
+            : this.copilotConfig.enterprise;
+
+        if (!orgName) {
+          throw new Error(`No ${type} name configured`);
+        }
+
+        const { data: installation } =
+          await appOctokit.rest.apps.getOrgInstallation({
+            org: orgName,
+          });
+
+        this.logger?.debug(
+          `[GithubClient] Got installation ID ${installation.id} for ${orgName}`,
+        );
+
+        // Now create the properly configured auth
+        octokitConfig.authStrategy = createAppAuth;
+        octokitConfig.auth = {
+          appId: authStrategy.appId,
+          privateKey: authStrategy.privateKey,
+          installationId: installation.id,
+        };
+      }
+
+      return new Octokit(octokitConfig);
+    } catch (error: any) {
+      this.logger?.error(
+        `[GithubClient] Failed to create Octokit instance for ${type}: ${
+          error?.message || error
+        }`,
+      );
+      throw await ResponseError.fromResponse(error?.response || error);
```

Comment:
I think we can't do this as ResponseError is only typically used for errors from fetch that returns are response that hasn't been consumed. If we want to map another type of error, it should be enough to throw a custom error having `status` or `statusCode` as property and the error handler will make sure to correctly map the error as HTTP response.

```suggestion
    } catch (error) {
      this.logger.error(
        `[GithubClient] Failed to create Octokit instance for ${type}: ${
          error?.message || error
        }`,
      );
      throw error;
```